src/cfnlint/data/ExtendedSpecs/all/03_value_types/aws_redshift.json

```diff
@@ -6,5 +6,45 @@
    "NumberMax": 100,
    "NumberMin": 1
   }
+ },
+ {
+  "op": "add",
+  "path": "/ValueTypes/AWS::Redshift::EndpointAccess.EndpointName",
+  "value": {
+   "StringMax": 2147483647,
+   "StringMin": 0
+  }
+ },
+ {
+  "op": "add",
+  "path": "/ValueTypes/AWS::Redshift::EndpointAccess.SubnetGroupName",
+  "value": {
+   "StringMax": 2147483647,
+   "StringMin": 0
+  }
+ },
+ {
+  "op": "add",
+  "path": "/ValueTypes/AWS::Redshift::EventSubscription.SubscriptionName",
+  "value": {
+   "StringMax": 2147483647,
```

Comment:
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-redshift-eventsubscription.html#cfn-redshift-eventsubscription-subscriptionname
While the printed text says the max is 2147483647 the conditions says the max is 255. I think additional fact finding may be needed on some of these.

I've found inconsistencies with different services in how the constraints are describe and quite often conflict with each other or omit a constraint and the api throws an error.

I would bet whoever wrote the docs for cloudformation just put the 32 bit signed int max and wrote in their docs more refined constraints. Like `AWS::Redshift::EventSubscription.SourceType` says it's max length is `2147483647` but it's a valid values are: cluster, cluster-parameter-group, cluster-security-group, cluster-snapshot, and scheduled-action.